File: src/entities/MovieRating.ts
import { Column, Entity, JoinColumn, ManyToOne } from "typeorm";
import { Movie } from "./Movie";
import { Users } from "./Users";

@Entity("movie_rating", { schema: "dionysus" })
export class MovieRating {
  @Column("integer", { name: "rating", nullable: true })
  rating: number | null;

  @ManyToOne(() => Movie, (movie) => movie.movieRatings, {
    onDelete: "CASCADE",
    onUpdate: "CASCADE",
  })
  @JoinColumn([{ name: "movie_id", referencedColumnName: "movieId" }])
  movie: Movie;

  @ManyToOne(() => Users, (users) => users.movieRatings, {
    onDelete: "CASCADE",
    onUpdate: "CASCADE",
  })
  @JoinColumn([{ name: "user_id", referencedColumnName: "userId" }])
  user: Users;
}

File: engine/engine.api/Models/MovieRating.cs
using System;
using System.Collections.Generic;

namespace engine.api.Models
{
    public partial class MovieRating
    {
        public long? MovieId { get; set; }
        public long? UserId { get; set; }
        public int? Rating { get; set; }

        public virtual Movie? Movie { get; set; }
        public virtual User? User { get; set; }
    }
}
